tools/ktx/image.hpp

```diff
@@ -45,125 +45,364 @@ template <typename S> inline S minimum(S a, S b) { return (a < b) ? a : b; }
+
+private:
     /* We're following what Netpbm does. This is their comment and code. */
 
     /* Here are parameters of the gamma transfer function for the Netpbm
-       formats.  This is ITU-R Recommendation BT.709, FKA CIE Rec 709.  It is
-       also ITU-R Recommendation BT.601, FKA CCIR 601.
-
-       This transfer function is linear for sample values 0 .. .018
-       and an exponential for larger sample values.
-       The exponential is slightly stretched and translated, though,
-       unlike the popular pure exponential gamma transfer function.
-
-       The standard actually defines the linear expansion as 4.500, which
-       means there is a discontinuity at linear intensity .018.  We instead
-       use ~4.514 to make a continuous function.  This may have been simply
-       a mistake when this code was written or based on an actual benefit
-       to having a continuous function -- The history is not clear.
-
-       Note that the discrepancy is below the precision of a maxval 255
-       image.
+    formats.  This is ITU-R Recommendation BT.709, FKA CIE Rec 709.  It is
+    also ITU-R Recommendation BT.601, FKA CCIR 601.
+
+    This transfer function is linear for sample values 0 .. .018
+    and an exponential for larger sample values.
+    The exponential is slightly stretched and translated, though,
+    unlike the popular pure exponential gamma transfer function.
+
+    The standard actually defines the linear expansion as 4.500, which
+    means there is a discontinuity at linear intensity .018.  We instead
+    use ~4.514 to make a continuous function.  This may have been simply
+    a mistake when this code was written or based on an actual benefit
+    to having a continuous function -- The history is not clear.
+
+    Note that the discrepancy is below the precision of a maxval 255
+    image.
     */
-    float const eoGamma = 2.2f;
-    float const oeGamma = 1.0f / eoGamma;
-    float const linearCutoff = 0.018f;
-    float const linearExpansion =
-        (1.099f * pow(linearCutoff, oeGamma) - 0.099f) / linearCutoff;
+    const float eoGamma_{2.2f};
+    const float oeGamma_{1.0f / eoGamma_};
+    const float linearCutoff_{0.018f};
+    const float linearExpansion_{(1.099f * pow(linearCutoff_, oeGamma_) - 0.099f) / linearCutoff_};
+};
 
-    float brightness;
+struct TransferFunctionBT2100_PQ_EOTF : public TransferFunction {
+    float decode(const float brightness) const override {
+        float Ym1 = pow(brightness, m1_);
+        return pow((c1_ + c2_ * Ym1) / (1 + c3_ * Ym1), m2_);
+    }
 
-    if (intensity < linearCutoff)
-        brightness = intensity * linearExpansion;
-    else
-        brightness = 1.099f * pow(intensity, oeGamma) - 0.099f;
+    float encode(const float intensity) const override {
+        float Erm2 = pow(intensity, rm2_);
+        return std::pow(std::max(Erm2 - c1_, 0.f) / (c2_ - c3_ * Erm2), m1_);
+    }
 
-    return brightness;
-}
+private:
+    const float m1_{0.1593017578125f};
```

Comment:
Please add some comments about what these constants are. At a minimum a reference to the spec. where they are defined.

Reply:
Each of these come from corresponding sections of the [Data Format Specification](https://registry.khronos.org/DataFormat/specs/1.3/dataformat.1.3.html).

Reply:
Please add a link to relevant section of the DF Spec.